sdk/python/kfp/_client.py

```diff
@@ -147,11 +147,8 @@ def _load_config(self, host, client_id, namespace, other_client_id, other_client
 
     host = host or ''
     # Preprocess the host endpoint to prevent some common user mistakes.
-    # This should only be done for non-IAP cases (when client_id is None). IAP requires preserving the protocol.
     if not client_id:
-      # Per feedback in proxy env, http or https is still required
-      if not proxy:
-        host = re.sub(r'^(http|https)://', '', host)
```

Comment:
I'm curious why we added it in the first place, if we can just remove it.

Reply:
originally removes http/s for all

later we find proxy can't work, so exclude it for proxy

now we exclude it for all.

Reply:
What was the original reason for removing http|https? Did that requirement change?

Reply:
Original PR: https://github.com/kubeflow/pipelines/pull/3287/files#diff-1f65a44a2187458da0f00e0ebbe730e5R167
Original issue: https://github.com/kubeflow/pipelines/issues/3280

I think we still need it, but we need to re-examine if this is compliant with the function `_is_inverse_proxy_host`

Reply:
> What was the original reason for removing http|https? Did that requirement change?

This is because inverse proxy has some problem when http(s) is provided as part of the host name.

Reply:
I tested https and it works well.

Reply:
So I guess http shouldn't work any way?
/approve